core/tests/behavior/copy.rs

```diff
@@ -105,7 +105,7 @@ pub async fn test_non_existing_source(op: Operator) -> Result<()> {
         .copy(&source_path, &target_path)
         .await
         .expect_err("copy must fail");
-    assert_eq!(err.kind(), ErrorKind::NotFound);
+    assert!(err.kind() == ErrorKind::NotFound || err.kind() == ErrorKind::PermissionDenied);
```

Comment:
Why we need to check this?

Reply:
When testing, I found that s3 returned a PermissionDenied error for a non-existent one.

![image](https://user-images.githubusercontent.com/24221472/230126727-a2641925-6cac-4bd9-9fd0-6eb90c76ff65.png)